plugins/woocommerce/src/Blocks/BlockTypes/StoreNotices.php

```diff
@@ -48,23 +48,62 @@ protected function render( $attributes, $content, $block ) {
 		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes, array(), array( 'extra_classes' ) );
 
 		return sprintf(
-			'<div %1$s>%2$s</div>',
+			'<div %1$s>%2$s%3$s</div>',
 			get_block_wrapper_attributes(
 				array(
 					'class' => 'wc-block-store-notices woocommerce ' . esc_attr( $classes_and_styles['classes'] ),
 				)
 			),
-			wc_kses_notice( $notices )
+			wc_kses_notice( $notices ),
+			$this->render_interactivity_notices_region()
 		);
 	}
 
 	/**
-	 * Get the frontend script handle for this block type.
-	 *
-	 * @param string $key Data to get, or default to everything.
+	 * In addition to the server notices we render interactivity API powered notices that
+	 * can be added client-side.
 	 */
-	protected function get_block_type_script( $key = null ) {
-		return null;
+	protected function render_interactivity_notices_region() {
+		$namespace = array( 'namespace' => 'woocommerce/store-notices' );
+
+		wc_initial_state(
+			'woocommerce/store-notices',
+			array(
+				'notices' => array(),
+			)
+		);
+
+		ob_start();
+		// Adding the is-layout-constrained class to the wrapper is a hack, to ensure the notice looks the same as it used to when prepended to post content.
+		?>
+		<div data-wc-interactive="<?php echo esc_attr( wp_json_encode( $namespace ) ); ?>" class="woocommerce-notices-wrapper">
```

Comment:
In our codebase we need to escape iAPI data with: 
```php
wp_json_encode( $interactive, JSON_NUMERIC_CHECK | JSON_HEX_TAG | JSON_HEX_APOS | JSON_HEX_QUOT | JSON_HEX_AMP ) 
```

For instance:

https://github.com/woocommerce/woocommerce/blob/4b987cfe4e5bc81c0a61e7126bcc4df5391745f5/plugins/woocommerce/src/Blocks/BlockTypes/ProductButton.php#L177-L180